pkg/monitoring/prometheusagent/config.go

```diff
@@ -44,10 +43,12 @@ func (pas PrometheusAgentService) buildRemoteWriteConfig(ctx context.Context,
 		"service_priority": getServicePriority(cluster),
 	}
 
-	shards, err := getShardsCountForCluster(ctx, cluster, currentShards)
+	// Compute the number of shards based on the number of series.
+	headSeries, err := querier.QueryTSDBHeadSeries(ctx, cluster.Name)
 	if err != nil {
-		return nil, errors.WithStack(err)
+		logger.Error(err, "failed to query head series")
 	}
+	shards.ComputeShards(currentShards, headSeries)
```

Comment:
Did you not forget to assign thé result of ComputeShards?

Reply:
Otherwise this looks good but do we want to increase an error counter hère so we Can get alerted if this happens for a long time?

Reply:
oops right I missed the assignation.

How to increase the error counter ?

Reply:
Either we still make it a réconciliation error but we ensure the création of the secret or you create a custom metric

Reply:
Added a new metric, because we should not return both a result and an error